packages/experiments/src/components/ExtendedPicker/BaseFloatingPicker.tsx

```diff
@@ -0,0 +1,486 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  KeyCodes,
+  autobind,
+  css,
+  getRTL
+} from '../../Utilities';
+import { Callout, DirectionalHint } from 'office-ui-fabric-react/lib/Callout';
+import { ValidationState, Suggestions, ISuggestionsProps, SuggestionsController, IBasePickerSuggestionsProps, ISuggestionModel }
+  from 'office-ui-fabric-react/lib/Pickers';
+import { IBaseFloatingPicker, IBaseFloatingPickerProps } from './BaseFloatingPicker.Props';
+import { IPersonaProps } from 'office-ui-fabric-react/lib/Persona';
+
+export interface IBaseFloatingPickerState {
+  queryString: string;
+  suggestedDisplayValue?: string;
+  moreSuggestionsAvailable?: boolean;
+  isSearching?: boolean;
+  isMostRecentlyUsedVisible?: boolean;
+  suggestionsVisible?: boolean;
+  suggestionsLoading?: boolean;
+  isResultsFooterVisible?: boolean;
+  didBind: boolean;
+}
+
+export class BaseFloatingPicker<T, P extends IBaseFloatingPickerProps<T>> extends BaseComponent<P, IBaseFloatingPickerState>
+  implements IBaseFloatingPicker {
+  protected selection: Selection;
+
+  protected root: HTMLElement;
+  protected suggestionElement: Suggestions<T>;
+
+  protected suggestionStore: SuggestionsController<T>;
+  protected SuggestionOfProperType: new (props: ISuggestionsProps<T>) => Suggestions<T> =
+  Suggestions as new (props: ISuggestionsProps<T>) => Suggestions<T>;
+  protected loadingTimer: number | undefined;
+  // tslint:disable-next-line:no-any
+  protected currentPromise: PromiseLike<any>;
+
+  constructor(basePickerProps: P) {
+    super(basePickerProps);
+
+    this.suggestionStore = new SuggestionsController<T>();
```

Comment:
I think that SuggestionsController should be a property passed into the picker. That way you can specify you're own behavior like multiple suggestion selection. 

Additionally it might make sense for suggestions controller to actually take over the responsibility of resolving new suggestions when text is entered which could mean that the onresolvesuggestions property is passed to suggestions. Although maybe that will come later.

Reply:
I've made the SuggestionsController as a prop to be passed into the floating picker. Let me know if this doesn't suffice.